src/ride/track.c

```diff
@@ -4880,15 +4883,456 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+    // and     ax, 0FFE0h
+	uint16 xRounded = floor2(x, 32);
+    // and     cx, 0FFE0h
+	uint16 yRounded = floor2(y, 32);
+
+    // push    edi
+    // push    esi
+
+	// movzx   di, byte ptr [esi+2]
+	// shl     di, 3
+	uint16 di = mapElement->base_height;
+	di = di << 3;
+
+    // movzx   si, byte ptr [esi+3]
+    // shl     si, 3
+	uint16 si = mapElement->clearance_height;
+	si = si << 3;
+
+    // call    map_invalidate_tile
+	map_invalidate_tile(xRounded, yRounded, mapElement->base_height * 8, mapElement->clearance_height * 8);
+
+	// pop     esi
+    // pop     edi
+
+	// mov     bx, [esi+5]
+	// and     bx, 8888h
+	// cmp     bx, 8888h
+	// jnz     short loc_6CDDFD
+	uint16 bx = mapElement->properties.track.maze_entry;
+	bx = bx & 0x8888;
+	if (bx - 0x8888 == 0) {
+		// call    map_element_remove
+		map_element_remove(mapElement);
+
+
+		// mov     dl, byte ptr [esp+14h+var_10]
+		// call    sub_6CB945
+		sub_6CB945(rideIndex);
+
+		// movzx   ebp, byte ptr [esp+14h+var_10]
+		// imul    ebp, 260h
+		// dec     ss:word_1362A56[ebp]
+		g_ride_list[rideIndex].maze_tiles--;
+	}
+
+    return loc_6CDDFD();
+}
+
+money32 loc_6CDDA0(uint16 x, uint16 y, uint8 rideIndex, rct_map_element *mapElement) {
+    // pop     esi
+    // pop     edx
+	return loc_6CDDA2(x, y, rideIndex, mapElement);
+}
+
+
+money32 loc_6CDB35(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z, rct_map_element *mapElement) {
+	if ((flags & GAME_COMMAND_FLAG_APPLY) == 0) {
+		return loc_6CDDFD();
+	}
+
+
+	if (mode == 1) {
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+	if (mode < 1) {
+		// loc_6CDCC9:
+		uint8 ax = x & 0x1F; // 0 - 31
+		uint8 cx = y & 0x1F; // 0 - 31
+
+		uint8 edx = 0b0011; // 3
+		uint8 bx = ax | cx;
+		if (bx != 0) {
+			edx = 0b1011; // 11
+			if ((cx & ax) == 0) {
+				edx = 0b0111; // 7
+
+				if (cx == 0) {
+					edx = 0b1111; // 15
+				}
+			}
+		}
+
+		// 0x6cdcff
+		mapElement->properties.track.maze_entry &= ~(1 << edx);
+
+		if (direction == 4) {
+			return loc_6CDDA2(x, y, rideIndex, mapElement);
+		}
+
+		// 0x6cdd12
+		edx = RCT2_GLOBAL(0x993ce9, uint8*)[direction + edx];
+		// TODO: wth is edx?
+		mapElement->properties.track.maze_entry &= ~(1 << edx);
+
+		// SAVE edx + esi
+
+		// TODO: btr     [esi+5], dx
+		// (esi+5 = mapElement->properties.track.maze_entry)
+
+		uint32 temp_edx = RCT2_GLOBAL(0x993cfc, uint8*)[edx];
+		if ((temp_edx & 0xFF) == 0xFF) {
+			return loc_6CDDA0(x, y, rideIndex, mapElement);
+		}
+
+		uint16 xRounded = floor2(x, 32);
+		uint16 yRounded = floor2(y, 32);
+
+		uint32 ebx = direction;
+		xRounded -= TileDirectionDelta[direction].x;
+		yRounded -= TileDirectionDelta[direction].y;
+
+		uint8 bh = rideIndex;
+		uint8 bl = (uint8) ((z >> 3) & 0xFF);
+
+		rct_map_element *temp_esi = map_get_first_element_at(xRounded / 32, yRounded / 32);
+
+		// 0x6cdd71:
+		do {
+			if (map_element_get_type(temp_esi) == MAP_ELEMENT_TYPE_TRACK) {
```

Comment:
Have a look at all of the other map element functions the general style for these is always

```
do
{
  if (tile type != type)
    continue;

  if (tile height != height)
    continue;

  if (tile ride Index != rideIndex)
    continue;

  tile found do something
}while (!last_tile);